src/ts/components/Controls.tsx

```diff
@@ -57,6 +62,43 @@ class Controls extends Component<IProps, IState> {
       ? "Sidebar__header-container"
       : "Sidebar__header-container Sidebar__header-container--is-closed";
 
+    const renderErrorMessage = () => {
+      const pluginState = this.state.pluginState;
+
+      if (!pluginState) {
+        return;
+      }
+
+      const hasAuthError = selectHasAuthError(pluginState);
+      const hasGeneralError = selectHasGeneralError(pluginState);
+      const hasErrors: boolean = hasAuthError || hasGeneralError;
+
+      let errorMessage: string = "";
+      if (hasAuthError) {
+        errorMessage = "Authentication error - please refresh the page. ";
+      }
+      if (hasGeneralError && !hasAuthError) {
+        errorMessage =
+          "Error fetching matches. Please try checking the document again. ";
+      }
+      if (hasErrors) {
```

Comment:
Could we return earlier here (e.g. L75), to save us trying to make up the error message when no errors exist?